pkg/sql/explain_distsql.go

```diff
@@ -227,11 +227,7 @@ func (n *explainDistSQLNode) startExec(params runParams) error {
 		}
 		diagram.AddSpans(spans)
 	} else {
-		nodeID, err := params.extendedEvalCtx.NodeID.OptionalNodeIDErr(distsql.MultiTenancyIssueNo)
-		if err != nil {
-			return err
-		}
-		flows := plan.GenerateFlowSpecs(nodeID)
+		flows := plan.GenerateFlowSpecs(roachpb.NodeID(params.extendedEvalCtx.NodeID.SQLInstanceID()))
```

Comment:
Casting to NodeID makes me uneasy on general principles as confusing them in general can lead to correctness problems. Can you get away with a zero NodeID here and elsewhere? Taking a step back, SQL flows for a tenant will never be scheduled on a "Node" in a multi-tenant world, it will always be a tenant. I don't think we necessarily want to do the code changes for that now (since we won't schedule nonlocal flows anyway, at least not for the time being), but it seems good to keep the concepts as separate as we can until then.

Reply:
I agree. the best way to go about this is to change `GenerateFlowSpecs` to work with SQL instance IDs, but that'll mean changing a lot of the planning code which doesn't seem worth to do at this stage. I'm not sure whether a 0 node ID would work.

My goal was to just get `EXPLAIN` working. I think having this cast is worth the cost of making that work. I should at least add a TODO here though. Do you prefer to hold off on merging this?

Reply:
No, this looks like a SQL team decision and that's you :-) Go ahead and merge.

Reply:
Added a TODO and a linked issue.